app/controllers/legacy_urls_controller.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class LegacyUrlsController < ApplicationController
+  # Handles legacy urls from Scholarsphere v3
+  def v3
+    uuid = LegacyIdentifier.find_uuid(version: 3, old_id: params[:id])
+    redirect_to resource_path(uuid)
```

Comment:
Code Climate thinks this is untested, but that's not true: ` spec/request/legacy_urls_spec.rb`